scripts/o3de/o3de/utils.py

```diff
@@ -162,6 +165,31 @@ def backup_folder(folder: str or pathlib.Path) -> None:
             if backup_folder_name.is_dir():
                 renamed = True
 
+def get_file_uri_from_git_api(parsed_uri):
+    """
+    Uses known git APIs to det a direct link to a file
+    :param parsed_uri: git uri with file appended
+    :return: direct link uri
+    """
+    if 'github.com' in parsed_uri.netloc:
+        components = parsed_uri.path.split('/')
+        components = [ele for ele in components if ele.strip()]
+
+        if len(components) < 3:
+            return parsed_uri
+
+        user = components[0]
+        repository = components[1].replace('.git','')
+        filepath = '/'.join(components[2:len(components)])
+        api_url = f'http://api.github.com/repos/{user}/{repository}/contents/{filepath}'
+
+        with urllib.request.urlopen(api_url) as url:
+            json_data = json.loads(url.read().decode())
+            download_url = json_data['download_url']
+            parsed_uri = urllib.parse.urlparse(download_url)
+        return parsed_uri
+    else:
+        return parsed_uri
```

Comment:
For API cleaniness, this should be separted into a github_utils.poy file and separated the general o3de package utils.
github isn't the only git storage provider and there is no guarantee that the O3DE foundation will not switch to other providers in the future(such as gitlab).

Reply:
Actually this can be combined into the same github specific python script that was suggested in the download.py.